File: src/main/java/com/digsol/turnomed/mapper/PatientMapper.java
package com.digsol.turnomed.mapper;

import com.digsol.turnomed.model.Patient;
import org.apache.ibatis.annotations.Mapper;
import org.springframework.stereotype.Component;

@Mapper
@Component
public interface PatientMapper {
    int deleteByPrimaryKey(Integer userId);

    int insert(Patient record);

    int insertSelective(Patient record);

    Patient selectByPrimaryKey(Integer userId);

    int updateByPrimaryKeySelective(Patient record);

    int updateByPrimaryKey(Patient record);
}

File: src/main/kotlin/com/digsol/turnomed/mapper/PatientMapper.kt
package com.digsol.turnomed.mapper

import com.digsol.turnomed.model.Patient
import org.apache.ibatis.annotations.Mapper
import org.springframework.stereotype.Component

@Mapper
@Component(value = "patientMapper")
interface PatientMapper {
    fun deleteByPrimaryKey(userId: Integer): Int

    fun insert(record: Patient): Int

    fun insertSelective(record: Patient): Int

    fun selectByPrimaryKey(userId: Integer): Patient

    fun updateByPrimaryKeySelective(record: Patient): Int

    fun updateByPrimaryKey(record: Patient): Int
}
